### ISSUE TITLE
Podman v2 doesn't work with podman-env

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#8596]
**If you want to build images using `podman`, you will have to use release 1.9.3**

The earlier releases (1.8.2) hangs, and the later releases (2.0.0) gives an error.

It _should_ also be OK to use the 1.6.x releases, available in many distributions.

The "varlink" socket is not supported anymore, so need to change minikube for it.

----

https://podman.io/blogs/2020/08/01/deprecate-and-remove-varlink-notice.html

> ... Podman API v1.0 for Podman is considered to be deprecated. If there are issues with the Podman API v1.0 in versions of Podman prior to v2.0 and those versions are still under support on Red Hat Enterprise Linux (RHEL), the Podman team will make a best effort to address those issues. However, no new feature requests for the API v1.0 will be considered and any problems found with the API v1.0 in Podman v2.0 will not be addressed.

> This also serves as a notification that the Podman v1.0 (varlink) API will be removed from the main GitHub branch of Podman in the near future. With the release of Podman v2.0 the Podman developers deprecated the Podman API v1.0 in favor of the new Podman v2.0 RESTful API. The plan is to remove varlink completely from the Podman v3.0 development branch which will be created some time after September 2020.

Original post: https://podman.io/blogs/2020/06/29/podman-v2-announce.html

**Comment by afbjorklund** [comment]
Binaries are available here: https://github.com/containers/libpod/releases/tag/v1.9.3

The packages in kubic repositories unfortunately only have the latest-and-greatest

**Comment by afbjorklund** [comment]
I was trying again with podman 2.0.5, but now there is not even an error.

It just fails silently... Trying to build over ssh (as a workaround also fails)

**Comment by afbjorklund** [comment]
The plan to replace sudo, is to make the socket owned by a local group:

https://github.com/containers/podman/issues/6809#issuecomment-651237669

And then make the default user part of this group (i..e same as for docker)

```bash
    mkdir -p /etc/systemd/system/podman.socket.d
    cat >/etc/systemd/system/podman.socket.d/override.conf <<EOF
[Socket]
SocketMode=0660
SocketUser=root
SocketGroup=podman
EOF
    systemctl daemon-reload
    echo "d /run/podman 0770 root podman" > /etc/tmpfiles.d/podman.conf
    systemd-tmpfiles --create
```

The original settings are in:

/usr/lib/systemd/system/podman.socket

```ini
[Unit]
Description=Podman API Socket
Documentation=man:podman-system-service(1)

[Socket]
ListenStream=%t/podman/podman.sock
SocketMode=0660

[Install]
WantedBy=sockets.target
```

/usr/lib/tmpfiles.d/podman.conf

```txt
d /run/podman 0700 root root
```

**Comment by afbjorklund** [comment]
https://podman.io/releases/2020/10/05/podman-release-v2.1.0.html

Since then podman 2.1.1 has been released to fix some bugs in 2.1.0....

**Comment by afbjorklund** [comment]
The new "podman-env" command works with both podman v1 and podman v2.

It checks the presence of the "varlink" command, in order to choose protocol...
